nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/http1/Http1ClientConnection.java

```diff
@@ -105,33 +110,50 @@ boolean isConnected() {
         return socket != null && socket.isConnected();
     }
 
+    private InetSocketAddress inetSocketAddress() {
+        DnsResolver dnsResolver = connectionKey.dnsResolver();
+        if (dnsResolver.useDefaultJavaResolver()) {
+            return new InetSocketAddress(connectionKey.host(), connectionKey.port());
+        } else {
+            InetAddress address = dnsResolver.resolveAddress(connectionKey.host(), connectionKey.dnsAddressLookup());
+            return new InetSocketAddress(address, connectionKey.port());
+        }
+    }
+
+    private int connectToProxy(InetSocketAddress remoteAddress) throws IOException {
```

Comment:
It would be better to handle this communication with an HttpClient, the problem is that I have to get the socket of it and pass it to this class for further processing. I am listening your suggestions.

Reply:
> Hm, just to clarify here. What to you mean by HttpClient CONNECT method? You mean Http1ClientConnection#connect?

I mean the HTTP method CONNECT. Similarly to GET, POST, PUT, etc, there is other HTTP method named CONNECT. This is the method that you use to connect to a server proxy.

Reply:
So you really meant CONNECT HTTP method. And now I have found out, we do not have it yet :-) It make sense now :D Let me think about this a bit.